pkg/resolver/search.go

```diff
@@ -508,49 +533,66 @@ func WhereClauseFilter(input *model.SearchInput) []exp.Expression {
 		}
 	}
 	if input.Filters != nil {
+
 		for _, filter := range input.Filters {
 			if len(filter.Values) > 0 {
 				values := pointerToStringArray(filter.Values)
 
-				// If property is of array type like label, remove the equal sign in it and use colon
-				// - to be similar to how it is stored in the database
-				if _, ok := arrayProperties[filter.Property]; ok {
-					cleanedVal := make([]string, len(values))
-					for i, val := range values {
-						labels := strings.Split(val, "=")
-						if len(labels) == 2 {
-							cleanedVal[i] = fmt.Sprintf(`{"%s":"%s"}`, labels[0], labels[1])
-						} else if len(labels) == 1 {
-							//// If property is of array type, format it as an array for easy searching
-							cleanedVal[i] = labels[0]
-						} else {
-							klog.Error("Error while decoding label string")
-							cleanedVal[i] = val
-						}
+				if dataTypeInMap, ok := propTypeMapFromCache[filter.Property]; !ok { //check if value exists/if value doesn't exist
+					klog.Warningf("Property type for [%s] doesn't exist property type map. Refreshing property type cache", filter.Property)
+					propTypeMap, _ := getPropertyTypeCache(ctx) //call database cache again
```

Comment:
Do we want to return the new updated map so that SearchResult resolver has the new map?